File: pubsub/src/main/java/glitch/pubsub/events/ChannelExtensionBroadcastEvent.java
package glitch.pubsub.events;

import com.google.gson.JsonArray;
import glitch.api.ws.events.IEvent;
import glitch.pubsub.GlitchPubSub;
import glitch.pubsub.Topic;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class ChannelExtensionBroadcastEvent extends AbstractMessageEvent<JsonArray> implements IEvent<GlitchPubSub> {
    public ChannelExtensionBroadcastEvent(GlitchPubSub client, Topic topic, JsonArray data) {
        super(client, topic, data);
    }
}

File: pubsub/src/main/kotlin/glitch/pubsub/events/ChannelExtensionBroadcastEvent.kt
package glitch.pubsub.events

import com.google.gson.JsonArray
import glitch.api.ws.events.IEvent
import glitch.pubsub.GlitchPubSub
import glitch.pubsub.Topic

class ChannelExtensionBroadcastEvent(client: GlitchPubSub, topic: Topic, data: JsonArray) : AbstractMessageEvent<JsonArray>(client, topic, data), IEvent<GlitchPubSub>
